gov2/demotools/pause.go

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package demotools provides a set of tools that help you write code examples.
+//
+// **Pausable**
+//
+// The pausable interface creates an easy to mock pausing object for testing.
+package demotools
+
+import "time"
+
+// IPausable defines the interface for pausable objects.
+type IPausable interface {
+	Pause(secs int)
+}
+
+// Pause holds the pausable object.
+type Pause struct{}
```

Comment:
I suppose this works, but it makes me uneasy to have a struct and func with the same name...maybe `Pausable` for the struct name?

Reply:
What about Pauser?